### PR TITLE
Feature/create way view

### PR DESCRIPTION
**Pull request by MaxDubchak**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/secure/RapidBoard.jspa?rapidView=777)

## Code reviewers

- [ ] @meelros
- [ ] @mizin4ik
- [x] @dimigor
- [ ] @PetrushynskyiOleksii
- [ ] @Sasha1152 
- [ ] @vo333ua 

### Second Level Review

- [ ] @lhalam

## Summary of change
Implemented Way View, some minor changes to way and place models, added ulrs for way view.

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: way_to_home/way/views.py [hunk 1]
```diff
@@ -1 +1,181 @@
...
+
+            user = request.user
+
+            ways = user.ways.all()
+            for way in ways:
+                data.append(way.get_way_with_routes())
+
+            return JsonResponse(data, status=200, safe=False)
+
+        way = Way.get_by_id(way_id)
```
Comment (dimigor) [inline way_to_home/way/views.py]: You must check if this way assigned to the user

File: way_to_home/way/views.py [hunk 2]
```diff
@@ -1 +1,181 @@
...
+                for request user if way_id is not specified
+                or HttpRequest with error if parameters are bad.
+        """
+        if not way_id:
+            data = []
+
+            user = request.user
+
+            ways = user.ways.all()
+            for way in ways:
```
Comment (dimigor) [inline way_to_home/way/views.py]: You may use the list comprehension

File: way_to_home/way/views.py [hunk 3]
```diff
@@ -1 +1,181 @@
...
+
+        user = request.user
+        if user.id != way.user_id:
+            return HttpResponse('Access denied for non-owner user', status=403)
+
+        data = {'name': data.get('name')}
+        # if not way_data_validator(data):
+        # return HttpResponse('Database operation has failed', status=400)
+
+        if not way.update(**data):
```
Comment (dimigor) [inline way_to_home/way/views.py]: maybe:
`is_updated = way.update(**data)`
`if not is_updated:`

**Review by dimigor — DISMISSED** [review]

File: way_to_home/way/views.py (hunk 1 above)
Comment (MaxDubchak) [reply way_to_home/way/views.py]: done

File: way_to_home/way/views.py (hunk 2 above)
Comment (MaxDubchak) [reply way_to_home/way/views.py]: done

File: way_to_home/way/views.py [hunk 4]
```diff
@@ -1 +1,194 @@
...
+                # Add validators
+                route = make_route_dict(step)
+                was_created = create_route(way, position, **route)
+
+                if not was_created:
+                    return HttpResponse('Bad request', status=400)
+                position += 1
+                routes.append(route)
+
+        return JsonResponse({'way': way.to_dict(),
```
Comment (lhalam) [inline way_to_home/way/views.py]: if not create way?

File: way_to_home/place/migrations/0001_initial.py [hunk 5]
```diff
@@ -21,7 +21,7 @@ class Migration(migrations.Migration):
                 ('longitude', models.DecimalField(decimal_places=6, max_digits=9)),
                 ('latitude', models.DecimalField(decimal_places=6, max_digits=9)),
                 ('name', models.CharField(blank=True, max_length=64)),
-                ('address', models.CharField(max_length=255)),
+                ('address', models.CharField(max_length=255, blank=True)),
```
Comment (lhalam) [inline way_to_home/place/migrations/0001_initial.py]: create new migrations

File: way_to_home/way/views.py [hunk 6]
```diff
@@ -1 +1,194 @@
...
+        # return HttpResponse('Database operation has failed', status=400)
+
+        is_updated = way.update(**data)
+        if not is_updated:
+            return HttpResponse('Database operation has failed', status=400)
+
+        return HttpResponse('Object was successfully updated', status=200)
+
+
+def make_route_dict(step):
```
Comment (lhalam) [inline way_to_home/way/views.py]: add _

File: way_to_home/way/views.py [hunk 7]
```diff
@@ -1 +1,194 @@
...
+            if not way:
+                return HttpResponse('Failed to create way', status=400)
+
+            routes = []
+            position = 0
+
+            for step in steps:
+                # Add validators
+                route = make_route_dict(step)
+                was_created = create_route(way, position, **route)
```
Comment (lhalam) [inline way_to_home/way/views.py]: implement method add rout

File: way_to_home/way/views.py [hunk 8]
```diff
@@ -1 +1,194 @@
...
+    route['time'] = step['duration']['value']
+
+    if step.get('transit_details'):
+        transport_id = step['transit_details']['line']['short_name']
+        route['transport_id'] = transport_id
+
+    return route
+
+
+def create_route(way, position, **kwargs):
```
Comment (lhalam) [inline way_to_home/way/views.py]: remove this method

**Review by PetrushynskyiOleksii — DISMISSED** [review]

**Review by dimigor — DISMISSED** [review]

**Review by PetrushynskyiOleksii — APPROVED** [review]

**Review by dimigor — APPROVED** [review]

**Pull request merged**